src/components/ErrorNotification/ErrorNotification.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+import { ErrorType } from '../../types/ErrorType';
+
+interface ErrorNotificationProps {
+  errorType: ErrorType | null;
+}
+
+export const ErrorNotification: React.FC<ErrorNotificationProps> = (
+  { errorType },
+) => {
+  const [visible, setVisible] = useState(true);
+
+  useEffect(() => {
+    setVisible(true);
+
+    if (errorType) {
+      const timerId = setTimeout(() => {
+        setVisible(false);
+      }, 3000);
+
+      return () => clearTimeout(timerId);
+    }
+
+    return () => { };
+  }, [errorType]);
+
+  const handleHideError = () => {
+    setVisible(false);
+  };
+
+  const getErrorMessage = () => {
+    switch (errorType) {
+      case ErrorType.LoadError:
+        return 'Unable to load todos';
+      case ErrorType.EmptyTitle:
+        return 'Title should not be empty';
+      case ErrorType.AddTodoError:
+        return 'Unable to add a todo';
+      case ErrorType.DeleteTodoError:
+        return 'Unable to delete a todo';
+      case ErrorType.UpdateTodoError:
+        return 'Unable to update a todo';
+      default:
+        return null;
+    }
+  };
+
+  return (
+    visible && errorType ? (
```

Comment:
actually it's better to use conditional rendering in parent component. This component is only responsible for rendering notification. if you move conditional rendering to parent component you can avoid setting timeout twice

Reply:
and btw you need only setting and deleting error message and you don't need second state visible. your error message will be shown only when error is present